### DISCUSSION TITLE
Single Fetch DTOs (Cannot encode object with prototype)

### DISCUSSION THREAD

**Discussion by ReptoxX** [remix-run/remix#9909] [Q&A]
As i'm using an SDK to talk to an external microservice for specific data. We use dtos and i can't just transmit them to the client if not wrapped in json().
As Single Fetch should make the use of json obsolete i'm wondering what i should try to fix the issue. `defer` and returning promises don't work either.

The error message i'm getting is: `Cannot encode object with prototype`.

**Comment by kiliman** [comment]
Resource routes can still return plain JSON objects. Only UI routes will encode the payload as `turbo-stream`.

**Reply by ReptoxX** [reply]
I'm sorry, that's not what i meant. I'm using an API for my UI-Routes for displaying information as the data doesn't come from remix. This api works using a SDK library with Request and Response Objects according to the swagger documentation. 
I just can't return these Objects to the UI to display because of the error above.

For some of those requests i want to implement a <Suspense> deferred response. Because some requests take a longer time and i don't want the customer to wait for all data.
